### ISSUE TITLE
Examples with Networking & Threading / async

### ISSUE THREAD

**Issue by masc-it** [pyedifice/pyedifice#185]
Hey,

First thing first, thanks for your work.

Is it possible to consider an example of app with networking requests? It's a pretty common scenario in real world apps. I would love to see the suggested patterns with pyedifice.

for example it would be great to tweak the financial example with data polled from some public endpoint (or dummy).

**Comment by jamesdbrock** [comment]
Yes it would be good to have an example with networking requests.

Even though we don't have a full example application, the basic technique for how to do this correctly is shown in the example code for the `use_async` Hook https://pyedifice.github.io/stubs/edifice.use_async.html#edifice.use_async

The only networking example we have currently is Financial Charting https://pyedifice.github.io/examples.html#financial-charting and that app does networking totally the wrong way, with blocking calls here.

https://github.com/pyedifice/pyedifice/blob/a1f756b688da6789bc5f4708360bf4c02ea4cdaf/examples/financial_charts.py#L199-L201

The correct way to do network requests is with `async` calls. I've been thinking about updating the Financial Charting app to use the `use_async` Hook with https://github.com/ttamg/yahoo-finance-async instead but I haven't done it yet.

**Comment by masc-it** [comment]
Got it, thank you for clarifying.

**Comment by masc-it** [comment]
Is this a good way to launch an async task on button click?

```
...
from my_pkg import net_request
...
data, data_set = ed.use_state("")

 async def _net_request():
        print("loading data...")
        data = await net_request("..")
        data_set(data)
        print("done")

fn_getdata, fn_getdata_cancel = ed.use_async_call(_net_request)
...
ed.Button("GET DATA", on_click=lambda _: fn_getdata())
```

**Comment by jamesdbrock** [comment]
I completely re-wrote the [Financial Charting Example](https://pyedifice.github.io/examples.html#financial-charting). Now it does `async` `ProcessPoolExecutor` calls to the normal synchronous [yfinance](https://pypi.org/project/yfinance/) API.

If your networking library has an `async` API then you don't need to use `ProcessPoolExecutor`.

**Comment by jamesdbrock** [comment]
Update actually it does `async` `ThreadPoolExecutor` calls to the normal synchronous [yfinance](https://pypi.org/project/yfinance/) API.
